### PR TITLE
Add grouped monthly/yearly subscription variants

### LINKED ISSUE #105: Add first-class monthly/yearly subscription variants with offer-family exclusivity
## Problem
We need the standard membership pattern where a customer sees one paid offer with a monthly/yearly toggle, can buy the yearly option at a discount (for example 10x monthly instead of 12x), and does not end up with parallel recurring grants for what is conceptually the same tier.

Today Perkcord models a tier as a single purchase type with a single checkout shape. That means monthly and yearly are only representable as separate tiers that happen to grant the same roles, which works mechanically but leaves product behavior undefined.

## Why this matters
- Clients expect a normal monthly/yearly toggle on the member-facing pricing page.
- Separate tiers create messy overlap behavior: a member can potentially hold both monthly and yearly recurring grants at the same time.
- Reporting, support, and future self-serve management become harder when billing cadence variants are modeled as unrelated products.
- We already identified this as a productization gap: `docs/productization-gaps.md:44`.

## Current state
- Tier schema is single-policy / single-config today: `convex/convex/schema.ts:131`, `convex/convex/entitlements.ts:21`.
- The admin tier editor exposes exactly one purchase type per tier (`subscription`, `one_time`, `lifetime`): `apps/web/components/admin/tier-editor.tsx:681`.
- Stripe can store multiple subscription price IDs on one tier, but checkout always picks the first one, so there is no member-facing cadence choice: `apps/web/lib/stripeCheckout.ts:23`, `apps/web/lib/stripeCheckout.ts:25`.
- Authorize.Net and NMI are effectively single-cadence per tier because checkout config is singular: `apps/web/lib/authorizeNetCheckout.ts:66`, `apps/web/lib/api/admin/tiersForm.ts:201`.
- Role sync is additive across all effective grants, so duplicate monthly + yearly subscriptions would union roles instead of enforcing exclusivity: `convex/convex/roleSync.ts:182`.
- Provider event processing creates or updates grants by provider object / source ref, but does not enforce recurring-variant exclusivity across related tiers: `convex/convex/providerEventProcessing.ts:320`, `convex/convex/providerEventProcessing.ts:427`.
- Free claims only prevent duplicate grants for the same exact tier; they do not define broader overlap policy: `apps/web/lib/api/subscribe/freeClaim.ts:102`.

## Goal
Add a first-class way to model recurring billing variants of the same offer so we can:
- present a single member-facing offer with a monthly/yearly toggle
- map each cadence to provider-specific checkout identifiers
- enforce a clear overlap policy for recurring variants of the same offer
- keep current additive entitlement behavior for unrelated offers unless configured otherwise

## Recommended product policy
Opinionated default for Phase 1:
- Monthly and yearly subscriptions are variants of the same paid offer family.
- Recurring variants within the same offer family are mutually exclusive.
- One-time, lifetime, and free entitlements remain additive by default, unless we later add explicit family rules for them.
- This issue should define initial purchase/select behavior and backend exclusivity foundations; full self-serve upgrade/downgrade management can layer on top later.

## Scope
- Extend tier or offer modeling to support:
  - shared offer family identity (for example `offerKey` / `tierFamilyId`)
  - explicit billing cadence / interval metadata (monthly, yearly)
  - optional family-level exclusivity for recurring variants
- Update admin tier configuration so monthly and yearly variants can be intentionally linked instead of created as unrelated tiers.
- Update public/member tier presentation so related recurring variants render as one offer with a cadence toggle.
- Ensure checkout resolves the selected cadence explicitly instead of silently picking the first configured provider ref.
- Define and implement the entitlement rule for an already-subscribed member selecting another recurring variant in the same family.
- Add tests for overlap handling, checkout selection, and provider-event mapping.

## Acceptance criteria
- Admins can configure a paid offer with at least two recurring variants: monthly and yearly.
- Member-facing pricing can show one logical offer with a monthly/yearly toggle.
- The selected cadence maps to the correct provider configuration for checkout.
- The system does not leave a member with parallel active monthly + yearly recurring entitlements for the same offer family.
- Unrelated tiers continue to behave additively unless explicitly grouped.
- One-time, lifetime, and free entitlements remain additive by default.
- Validation and tests cover ambiguous mappings and overlap-policy edge cases.

## Out of scope
- Full provider-native proration across Stripe / Authorize.Net / NMI.
- General-purpose limits like "max N subscriptions across the whole guild".
- A complete self-serve plan-switching UI in the member account area.
- Redesigning all non-recurring tier semantics beyond the default additive policy above.

## Related work
- `#61` member self-service cancellation / account management
- `#77` billing cancellation and account recovery when Discord access is lost
- `#68` / PR `#69` richer direct tier landing page
- `#79` / PR `#98` subscribe-flow session and member UX cleanup

## Open questions
- Do we want the offer-family concept to be generic enough for future free/one-time/lifetime grouping, or only for recurring variants initially?
- When switching between monthly and yearly later, should Phase 1 behavior be "start new cadence and deactivate prior family grant" or "disallow if an active family subscription already exists until explicit management flows ship"?
- Should exclusivity live purely in entitlement processing, or should we also add pre-checkout UI/validation once a member already has an active family subscription?

### PR DESCRIPTION
**Pull request by BASIC-BIT**
## Summary
- add first-class subscription offer-family metadata so monthly and yearly variants can be modeled, validated, and kept exclusive in the entitlement layer
- extend admin tier management plus REST plumbing to configure grouped recurring variants without changing one-time/lifetime behavior
- build grouped member pricing UI on top of the merged tier landing work, including smoke/visual coverage for the monthly/yearly selector

## Testing
- npm run lint
- npm run typecheck
- npx vitest run tests/entitlements.test.ts tests/providerEventProcessing.test.ts
- npx vitest run lib/api/admin/tiersForm.test.ts lib/tierCatalog.test.ts lib/subscriptionOffers.test.ts
- npm exec -- playwright test e2e/smoke.spec.ts --grep "grouped subscription offers"
- npm exec -- playwright test e2e/visual.spec.ts --grep "grouped subscription offer card"
- docker-based Linux snapshot update for `subscribe-grouped-offer-card`

Closes #105

### CODE REVIEW

**Comment by BASIC-BIT** [comment]
@perkcord preview

File: apps/web/components/admin/tier-editor.tsx [hunk 1]
```diff
@@ -380,6 +406,73 @@ function TierPolicyFields({
             If enabled, access stays active until the current paid period ends after cancellation.
           </FieldHint>
         </label>
+        <input
+          type="hidden"
+          name="clearSubscriptionOffer"
+          value={shouldClearSubscriptionOffer ? "true" : "false"}
+        />
```
[the remark this answers is not in the document]
Comment (BASIC-BIT) [reply apps/web/components/admin/tier-editor.tsx]: Moved the hidden `clearSubscriptionOffer` input to the always-rendered form root, so grouped subscription tiers can switch to one-time/lifetime without dropping the clear flag.
